modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/_diffs/plugins/addimages/plugin.js

```diff
@@ -167,6 +219,58 @@
 					this._processFile(imageFile, event.editor);
 				}
 			}
+			else if (event.data && event.data.type === 'html') {
+				const editor = event.editor;
+				const fragment = CKEDITOR.htmlParser.fragment.fromHtml(
+					event.data.dataValue
+				);
+
+				fragment.forEach((element) => {
+					if (
+						element.type === CKEDITOR.NODE_ELEMENT &&
+						element.name === 'img'
+					) {
+						const base64Regex = /^data:image\/(.*);base64,/;
+						const match = element.attributes.src.match(base64Regex);
+
+						if (element.attributes.src && match) {
+							const src = element.attributes.src;
+							const extension = match[1];
+							const name = `${Date.now().toString()}.${extension}`;
+
+							// Mark this "src" attribute as something we want to replace
+
+							this._imageSrcToReplace.push(src);
+
+							this._fetchBlob(src)
```

Comment:
We're doing this fetch to convert the base64 string (i.e. `data:....`) to a `Blob`.
If there's a better way to do it, I'm all for it

Reply:
Did you try [`new Blob(bin)`](https://javascript.info/blob)?

Reply:
Yes of course and:
1. That's not going to work because the first argument shouldn't be a string
2. The mime type is needed is you want a proper `Blob` object
3. How do you calculate the size ?

Reply:
So if you try what you suggest, here's what you'll see 

```javascript

const fakeBlob = new Blob([src]);
console.log(fakeBlob);    // Will output something Blob {size: 20434, type: ""} and size will depend on what you pasted of course
```

If you use `fetchBlob` with the same input string you'll not only see that the size is different, but that there's also a valid type... You could write a function that converts a string to a blob but I'm sure it will be more complex that doing a simple fetch and it might not work.

Reply:
[Quick search](https://stackoverflow.com/a/42508185/2103996) seems to indicate that `fetch` or Canvas or the two main ways?

Reply:
Fair enough, :+1: for `fetch`